File: 351004/Student/Lab6/publisher/src/main/java/com/example/task310/dto/WriterRequestTo.java
package com.example.task310.dto;

import com.example.task310.enums.WriterRole;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class WriterRequestTo {
    private Long id;
    @Size(min = 2, max = 64)
    private String login;
    @Size(min = 8, max = 128)
    private String password;
    @Size(min = 2, max = 64)
    private String firstname;
    @Size(min = 2, max = 64)
    private String lastname;
    private WriterRole role;
}

File: 351003/Soika/entities-app/publisher/src/main/kotlin/com/example/entitiesapp/dto/WriterRequestTo.kt
package com.example.entitiesapp.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size
import java.io.Serializable

data class WriterRequestTo(
    val id: Long? = null,

    @field:NotBlank
    @field:Size(min = 2, max = 64)
    val login: String,

    @field:NotBlank
    @field:Size(min = 8, max = 128)
    val password: String,

    @field:NotBlank
    @field:Size(min = 2, max = 64)
    val firstname: String,

    @field:NotBlank
    @field:Size(min = 2, max = 64)
    val lastname: String,

    val role: Role = Role.CUSTOMER
) : Serializable